examples/test-exec-doc.yml

```diff
@@ -0,0 +1,153 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
+  pull_request:
+    paths:
+      - '**/*.md'           # Trigger on pull requests affecting Markdown files
+  workflow_dispatch:        # Allows manual triggering of the workflow
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest  # Run the job on the latest Ubuntu runner
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0      # Fetch all history for all branches and tags
+
+    - name: Azure CLI Login
+      uses: azure/login@v1
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}             # Azure Client ID from secrets
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}             # Azure Tenant ID from secrets
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }} # Azure Subscription ID from secrets
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x' # Specify Python version 3.x
+
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip   # Upgrade pip
+        pip install PyGithub pyyaml # Install required Python packages
+
+    - name: Run tests and handle PR
+      env:
+        GITHUB_TOKEN: ${{ secrets.PAT }} # Personal Access Token from secrets
+      run: |
+        python <<EOF
+        import os
+        import subprocess
+        import shutil
+        import github
+
+        # Get GitHub token from environment variable
+        GITHUB_TOKEN = os.getenv('GITHUB_TOKEN')
+        # Authenticate with GitHub using PyGithub
+        g = github.Github(GITHUB_TOKEN)
+
+        # Check if 'ie' (Innovation Engine) is installed, install if not
+        if not shutil.which("ie"):
+            # Update package lists and install unzip
+            subprocess.run("sudo apt-get update && sudo apt-get install -y unzip", shell=True, check=True)
+            # Install Innovation Engine CLI
+            subprocess.run("curl -Lks https://aka.ms/install-ie | /bin/bash", shell=True, check=True)
```

Comment:
I would pin this to a specific version of IE, there's no guarantee that this example might work for future versions.

Reply:
can you deploy the release with the spinner not showing up in the gh action? Currently, I have it pinned to v0.2.3 and will update it once you have the latest version installed

Reply:
This work hasn't been done yet, but `v0.3.0` won't come out for a little bit because I'm including a v1 for pre-requisites in that release.

Reply:
so how would you remove the spinner for ignite?

Reply:
I won't, it's too late of an ask

Reply:
Let's put it for the future though - it is needed in any case